scripts/abundance-dist-single.py

```diff
@@ -68,116 +46,16 @@ def main():  # pylint: disable=too-many-locals,too-many-branches
-                                  args.n_tables)
-
-    print >>sys.stderr, 'kmer_size:', counting_hash.ksize()
-    print >>sys.stderr, 'k-mer counting table sizes:', \
-        counting_hash.hashsizes()
-    print >>sys.stderr, 'outputting to', args.output_histogram_filename
-
-    khmer.get_config().set_reads_input_buffer_size(args.threads * 64 * 1024)
-
-    # start loading
-    rparser = khmer.ReadParser(args.input_sequence_filename, args.threads)
-    threads = []
-    print >>sys.stderr, 'consuming input, round 1 --', \
-        args.input_sequence_filename
-    for _ in xrange(args.threads):
-        thread = \
-            threading.Thread(
-                target=counting_hash.consume_fasta_with_reads_parser,
-                args=(rparser, )
-            )
-        threads.append(thread)
-        thread.start()
-
-    for thread in threads:
-        thread.join()
-
-    if args.report_total_kmers:
-        print >> sys.stderr, 'Total number of unique k-mers: {0}'.format(
-            counting_hash.n_unique_kmers())
-
-    abundance_lists = []
-
-    def __do_abundance_dist__(read_parser):
-        abundances = counting_hash.abundance_distribution_with_reads_parser(
-            read_parser, tracking)
-        abundance_lists.append(abundances)
-
-    print >>sys.stderr, 'preparing hist from %s...' % \
-        args.input_sequence_filename
-    rparser = khmer.ReadParser(args.input_sequence_filename, args.threads)
-    threads = []
-    print >>sys.stderr, 'consuming input, round 2 --', \
-        args.input_sequence_filename
-    for _ in xrange(args.threads):
-        thread = \
-            threading.Thread(
-                target=__do_abundance_dist__,
-                args=(rparser, )
-            )
-        threads.append(thread)
-        thread.start()
-
-    for thread in threads:
-        thread.join()
-
-    assert len(abundance_lists) == args.threads, len(abundance_lists)
-    abundance = {}
-    for abundance_list in abundance_lists:
-        for i, count in enumerate(abundance_list):
-            abundance[i] = abundance.get(i, 0) + count
-
-    total = sum(abundance.values())
-
-    if 0 == total:
-        print >> sys.stderr, \
-            "ERROR: abundance distribution is uniformly zero; " \
-            "nothing to report."
-        print >> sys.stderr, "\tPlease verify that the input files are valid."
-        sys.exit(1)
-
-    sofar = 0
-    for _, i in sorted(abundance.items()):
-        if i == 0 and not args.output_zero:
-            continue
-
-        sofar += i
-        frac = sofar / float(total)
-
-        print >> hist_fp, _, i, sofar, round(frac, 3)
-
-        if sofar == total:
-            break
-
-    if args.savetable:
-        print >>sys.stderr, 'Saving k-mer counting table ', args.savetable
-        print >>sys.stderr, '...saving to', args.savetable
-        counting_hash.save(args.savetable)
+    abund_dist_single.do_abund_dist_single(args.input_sequence_filename,
```

Comment:
Break up abund_dist_single into reusable components.

What is the core of the script? Creating an abundance histogram from a counting table object and ReadParser object.

Create a python method that does that and returns the result. Move this method into Oxli.